rpc/legacyrpc/methods.go

```diff
@@ -1456,33 +1614,36 @@ func getWalletFee(icmd interface{}, w *wallet.Wallet) (interface{}, error) {
 var helpDescs map[string]string
 var helpDescsMu sync.Mutex // Help may execute concurrently, so synchronize access.
 
-// helpWithChainRPC handles the help request when the RPC server has been
-// associated with a consensus RPC client.  The additional RPC client is used to
-// include help messages for methods implemented by the consensus server via RPC
-// passthrough.
-func helpWithChainRPC(icmd interface{}, w *wallet.Wallet, chainClient *dcrrpcclient.Client) (interface{}, error) {
-	return help(icmd, w, chainClient)
-}
-
-// helpNoChainRPC handles the help request when the RPC server has not been
-// associated with a consensus RPC client.  No help messages are included for
-// passthrough requests.
-func helpNoChainRPC(icmd interface{}, w *wallet.Wallet) (interface{}, error) {
-	return help(icmd, w, nil)
-}
-
 // help handles the help request by returning one line usage of all available
 // methods, or full help for a specific method.  The chainClient is optional,
 // and this is simply a helper function for the HelpNoChainRPC and
 // HelpWithChainRPC handlers.
-func help(icmd interface{}, w *wallet.Wallet, chainClient *dcrrpcclient.Client) (interface{}, error) {
+func help(s *Server, icmd interface{}) (interface{}, error) {
 	cmd := icmd.(*dcrjson.HelpCmd)
+	// The "help" RPC must use an HTTP POST client when calling down to dcrd
+	// for additional help methods.  This is required to avoid including
+	// websocket-only requests in the help, which are not callable by wallet
+	// JSON-RPC clients.  Any errors creating the POST client may be ignored
+	// since the client is not necessary for the request.
+
+	chainClient, ok := s.requireChainClient()
+	if !ok {
```

Comment:
this check isn't necessary. In the old code, there were two entrypoint handlers for the help RPC because the dcrd RPC client was optional.  The ok check can be removed because the code further below asserts that chainClient is non-nil before using it, but a comment should be added here explaining why the check isn't needed here.